ee/packages/federation-matrix/src/services/MatrixMediaService.ts

```diff
@@ -0,0 +1,148 @@
+import type { HomeserverServices } from '@hs/federation-sdk';
+import { Upload } from '@rocket.chat/core-services';
+import type { IUpload } from '@rocket.chat/core-typings';
+import { Logger } from '@rocket.chat/logger';
+import { Uploads } from '@rocket.chat/models';
+
+const logger = new Logger('federation-matrix:media-service');
+
+export interface IRemoteFileReference {
+	name: string;
+	size: number;
+	type: string;
+	mxcUri: string;
+	serverName: string;
+	mediaId: string;
+}
+
+export class MatrixMediaService {
+	private static homeserverServices: HomeserverServices;
+
+	static setHomeserverServices(services: HomeserverServices): void {
+		this.homeserverServices = services;
+	}
+
+	static generateMXCUri(fileId: string, serverName: string): string {
+		return `mxc://${serverName}/${fileId}`;
+	}
+
+	static parseMXCUri(mxcUri: string): { serverName: string; mediaId: string } | null {
+		const match = mxcUri.match(/^mxc:\/\/([^/]+)\/(.+)$/);
+		if (!match) {
+			logger.error('Invalid MXC URI format', { mxcUri });
+			return null;
+		}
+		return {
+			serverName: match[1],
+			mediaId: match[2],
+		};
+	}
+
+	static async prepareLocalFileForMatrix(fileId: string, serverName: string): Promise<string> {
+		try {
+			const file = await Uploads.findOneById(fileId);
+			if (!file) {
+				logger.error(`File ${fileId} not found in database`);
+				throw new Error(`File ${fileId} not found`);
+			}
+
+			if (file.federation?.mxcUri) {
+				return file.federation.mxcUri;
+			}
+
+			const mxcUri = this.generateMXCUri(fileId, serverName);
+
+			await Uploads.setFederationInfo(fileId, {
+				mxcUri,
+				serverName,
+				mediaId: fileId,
+			});
+
+			return mxcUri;
+		} catch (error) {
+			logger.error('Error preparing file for Matrix:', error);
+			throw error;
+		}
+	}
+
+	static async getLocalFileForMatrixNode(mediaId: string): Promise<IUpload | null> {
+		try {
+			let file = await Uploads.findByFederationMediaId(mediaId);
+
+			if (!file) {
+				file = await Uploads.findOneById(mediaId);
+			}
```

Comment:
is this fallback really a thing? what is the situation a `mediaId` can be saved as the file `_id`?